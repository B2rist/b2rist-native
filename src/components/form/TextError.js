import {Text} from 'react-native-paper';
import React from 'react';
import PropTypes from 'prop-types';
import {StyleSheet} from 'react-native';
import {useTranslation} from 'react-i18next';

const TextError = ({error, touched}) => {
  const {t} = useTranslation();
  return (
    touched &&
    error && (
      <Text style={styles.error} variant="labelSmall">
        {t(error)}
      </Text>
    )
  );
};

const styles = StyleSheet.create({
  error: {
    color: 'red',
    marginLeft: 5,
  },
});

export default TextError;

TextError.propTypes = {
  error: PropTypes.string,
  touched: PropTypes.bool,
};

TextError.defaultProps = {
  error: null,
  touched: false,
};
