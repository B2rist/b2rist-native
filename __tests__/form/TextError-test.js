import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TextError from 'components/form/TextError';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const prepare = props => renderer.create(<TextError {...props} />).toJSON();

describe('text error tests', () => {
  it('test error not renders with no error', () => {
    const testError = prepare();
    expect(testError).toBeNull();
  });

  it('test error not renders when untouched', () => {
    const testError = prepare({error: 'error'});
    expect(testError).toBeNull();
  });

  it('test error renders when touched and error message', () => {
    const testError = prepare({error: 'error', touched: true});
    expect(testError).toMatchSnapshot();
  });
});
