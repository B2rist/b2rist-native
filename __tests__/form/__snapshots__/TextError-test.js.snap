// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text error tests test error renders when touched and error message 1`] = `
<Text
  style={
    [
      {
        "color": "rgba(28, 27, 31, 1)",
        "fontFamily": "System",
        "fontSize": 11,
        "fontWeight": "500",
        "letterSpacing": 0.5,
        "lineHeight": 16,
      },
      {
        "textAlign": "left",
      },
      {
        "writingDirection": "ltr",
      },
      {
        "color": "red",
        "marginLeft": 5,
      },
    ]
  }
>
  error
</Text>
`;
